File: app/src/main/java/com/lcl6/cn/basedialog/mvp/contract/ManagerContract.java
package com.lcl6.cn.basedialog.mvp.contract;

import com.lcl6.cn.component.base.mvp.presnenter.BasePresenter;
import com.lcl6.cn.component.base.mvp.view.BaseView;

/**
 * Created by liancl on 2017/9/1.
 */

public interface ManagerContract {
    interface Presenter extends BasePresenter<View>{
        void requestTabaoData();
        void setGlobal(String newUrl);
        void removeGlobal();
        void requestGuthub( String newUrl);
        void requestGank(String newUrl);
        void requestDouban(String newUrl);
    }

    interface View extends BaseView{
        void requestTabaoData(String message);
        void setGlobal(String message);
        void removeGlobal(String message);
        void requestGuthub( String message);
        void requestGank(String message);
        void requestDouban(String message);
    }
}

File: app/src/main/java/com/lcl6/cn/basedialog/mvp/contract/ManagerContract.kt
package com.lcl6.cn.basedialog.mvp.contract

import com.lcl6.cn.component.base.mvp.presnenter.BasePresenter
import com.lcl6.cn.component.base.mvp.view.BaseView

/**
 * Created by liancl on 2017/9/1.
 */

interface ManagerContract {
    interface Presenter : BasePresenter<View> {
        fun requestTabaoData()
        fun setGlobal(newUrl: String)
        fun removeGlobal()
        fun requestGuthub(newUrl: String)
        fun requestGank(newUrl: String)
        fun requestDouban(newUrl: String)
    }

    interface View : BaseView {
        fun requestTabaoData(message: String)
        fun setGlobal(message: String)
        fun removeGlobal(message: String)
        fun requestGuthub(message: String)
        fun requestGank(message: String)
        fun requestDouban(message: String)
    }
}
